back/src/domains/convention/adapters/PgConventionQueries.ts

```diff
@@ -116,6 +117,44 @@ export class PgConventionQueries implements ConventionQueries {
     );
   }
 
+  public async getConventionsMarketingData({
+    siret,
+  }: {
+    siret: SiretDto;
+  }): Promise<ConventionMarketingData[]> {
+    const result = await this.transaction
+      .selectFrom("conventions")
+      .innerJoin(
+        "actors as er",
+        "er.id",
+        "conventions.establishment_representative_id",
+      )
+      .select("conventions.siret")
+      .select("conventions.date_validation")
+      .select("conventions.date_end")
+      .select("er.email as establishment_representative_email")
+      .select("er.first_name as establishment_representative_firstname")
+      .select("er.last_name as establishment_representative_lastname")
+      .select("conventions.establishment_number_employees")
```

Comment:
tu peux faire un array tu sais :


```suggestion
.select([
  "conventions.siret",
  "conventions.date_validation",
  "conventions.date_end",
  "er.email as establishment_representative_email",
  "er.first_name as establishment_representative_firstname",
  "er.last_name as establishment_representative_lastname",
  "conventions.establishment_number_employees",
  ])
```

Reply:
et en fait tu peux même t'épargner le map en dessous en faisant ça :

```ts
select(eb => [
  "conventions.siret as siret",
  "conventions.date_validation as dateValidation",
  "conventions.date_end as dateEnd",
  jsonBuildObject({
    email: eb.ref("er.email"),
    firstName: eb.ref("er.first_name"),
    lastName: eb.ref("er.last_name"),
  }).as("establishmentRepresentative"),
  "conventions.establishment_number_employees as establishmentNumberEmployeesRange"
])
```